File: gridView/app/src/main/java/com/example/renanalves/gridview/MainActivity.java
package com.example.renanalves.gridview;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.GridLayout;
import android.widget.GridView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends Activity{

    GridView grid;
    List<Produto> produtos = new ArrayList<Produto>();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Produto produto = new Produto("Gol", R.drawable.gol,"Modelo 1.0 0km.");
        produtos.add(produto);
        produto = new Produto("Uno", R.drawable.uno2,"Modelo 1.0 10 mil km.");
        produtos.add(produto);
        produto = new Produto("Palio", R.drawable.palio,"Modelo 1.0 20 mil km.");
        produtos.add(produto);
        produto = new Produto("Gol2", R.drawable.gol,"Modelo 1.6 65 mil km.");
        produtos.add(produto);
        produto = new Produto("Uno2", R.drawable.uno2,"Modelo 1.4 30 mil km.");
        produtos.add(produto);
        produto = new Produto("Palio2", R.drawable.palio,"Modelo 1.4 45 mil km.");
        produtos.add(produto);

        grid = findViewById(R.id.gridView);
        final CustomGrid adapter = new CustomGrid(this, produtos);

        grid.setAdapter(adapter);

        final Intent intent = new Intent(this,ProdutoInformation.class);

//        grid.setOnItemClickListener(new AdapterView.OnItemClickListener(){
//
//            @Override
//            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
//                Produto contato = (Produto) adapterView.getAdapter().getItem(i);
//
//                intent.putExtra("Produto",contato);
//                startActivity(intent);
//            }
//        });
    }

}

File: EmprestimoApp/app/src/main/java/com/example/renanalves/emprestimoapp/MainActivity.kt
package com.example.renanalves.emprestimoapp

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.design.widget.NavigationView
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.widget.ListView
import com.example.renanalves.emprestimoapp.DataBase.DBHelper
import com.example.renanalves.emprestimoapp.Model.Emprestimo
import com.example.renanalves.emprestimoapp.Model.EmprestimoAdapter
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.app_bar_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    internal lateinit var db: DBHelper
    internal var emprestimos: List<Emprestimo> = ArrayList<Emprestimo>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        db = DBHelper(this)
        setupListView()

        listViewEmprestimos.setOnItemClickListener { parent, view, position, id ->
            val intent = Intent(this, DetailEmprestimoActivity::class.java)
            intent.putExtra("emprestimo",emprestimos[position])
            startActivity(intent)
        }

        fab.setOnClickListener {
            var intent = Intent(this,AddEmprestimoActivity::class.java)
            intent.putExtra("id",emprestimos.count())
            startActivity(intent)
            finish()
        }

        val toggle = ActionBarDrawerToggle(
            this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close
        )
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()

        nav_view.setNavigationItemSelectedListener(this)
    }

    private fun setupListView() {
        emprestimos = db.allEmprestimos

        var emprestimosAdapter = EmprestimoAdapter(this, emprestimos)
        var listViewEmprestimos = findViewById<ListView>(R.id.listViewEmprestimos)
        listViewEmprestimos.adapter = emprestimosAdapter
    }

    override fun onBackPressed() {
        if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
            drawer_layout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        when (item.itemId) {
            R.id.action_settings -> return true
            else -> return super.onOptionsItemSelected(item)
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        when (item.itemId) {
            R.id.nav_camera -> {
                // Handle the camera action
            }
//            R.id.nav_gallery -> {
//
//            }
//            R.id.nav_slideshow -> {
//
//            }
//            R.id.nav_manage -> {
//
//            }
//            R.id.nav_share -> {
//
//            }
//            R.id.nav_send -> {
//
//            }
        }
        drawer_layout.closeDrawer(GravityCompat.START)
        return true
    }
}
